server/user-bootstrap/index.js

```diff
@@ -13,25 +13,44 @@ import crypto from 'crypto';
 import { filterUserObject } from 'lib/user/shared-utils';
 import { getActiveTestNames } from 'lib/abtest/utility';
 import config from 'config';
+import { setSupportSession } from '../support-session';
 
 const debug = debugFactory( 'calypso:bootstrap' ),
 	API_KEY = config( 'wpcom_calypso_rest_api_key' ),
 	AUTH_COOKIE_NAME = 'wordpress_logged_in',
 	/**
 	 * WordPress.com REST API /me endpoint.
 	 */
+	SUPPORT_SESSION_API_KEY = config( 'wpcom_calypso_support_session_rest_api_key' ),
 	API_PATH = 'https://public-api.wordpress.com/rest/v1/me',
 	apiQuery = {
 		meta: 'flags',
 		abtests: getActiveTestNames( { appendDatestamp: true, asCSV: true } ),
 	},
 	url = `${ API_PATH }?${ stringify( apiQuery ) }`;
 
-module.exports = function( authCookieValue, geoCountry ) {
+/**
+ * Requests the current user for user bootstrap.
+ *
+ * @param {(string|undefined)} authCookieValue The authentication cookie, if there is one.
+ * @param {string}             geoCountry      The GeoIP country code.
+ * @param {(string|undefined)} supportSession  The support session.
+ *
+ * @returns {Promise<object>} A promise for a user object.
+ */
+module.exports = function( authCookieValue, geoCountry, supportSession ) {
```

Comment:
I was on the fence about this when making the update. It's easier to just pass the request, and maybe you could say it's user bootstrap's business what it needs from the request anyway. But I opted for explicit arguments so the requirements of user bootstrap would be clearer. They aren't 100% clear though because the function needs either the login cookie or the support session ID, not both.

I'll move to the single request argument.